### PR TITLE
Algo_Registry: use critical section for locking on Windows

### PR DESCRIPTION
**Pull request by matejk**
#297: Fix for deadlock in Algo_Registry when loading botan.dll: use critical section instead of std::mutex.

### CODE REVIEW

File: src/lib/filters/comp_filter.cpp [hunk 1]
```diff
@@ -25,7 +25,12 @@ Compression_Decompression_Filter::Compression_Decompression_Filter(Transform* tr
    {
    m_transform.reset(dynamic_cast<Compressor_Transform*>(transform));
    if(!m_transform)
-      throw std::invalid_argument("Transform " + transform->name() + " is not a compressor");
+      {
+      if (transform)
+         throw std::invalid_argument("Transform " + transform->name() + " is not a compressor");
+      else
+         throw std::invalid_argument("Transform is null");
+      }
```
Comment (webmaster128) [inline src/lib/filters/comp_filter.cpp]: Could you move this part into a separate PR please? This might get in more easily since it is quickly reviewed.

**Comment by matejk** [comment]
Created pull requests #320 and #321.

Closing this one.
